pkg/promlib/querydata/request.go

```diff
@@ -88,22 +88,49 @@ func (s *QueryData) Execute(ctx context.Context, req *backend.QueryDataRequest)
 		Responses: backend.Responses{},
 	}
 
-	cfg := backend.GrafanaConfigFromContext(ctx)
-	hasPromQLScopeFeatureFlag := cfg.FeatureToggles().IsEnabled("promQLScope")
-	hasPrometheusDataplaneFeatureFlag := cfg.FeatureToggles().IsEnabled("prometheusDataplane")
+	var (
+		cfg                               = backend.GrafanaConfigFromContext(ctx)
+		hasPromQLScopeFeatureFlag         = cfg.FeatureToggles().IsEnabled("promQLScope")
+		hasPrometheusDataplaneFeatureFlag = cfg.FeatureToggles().IsEnabled("prometheusDataplane")
+		hasPrometheusRunQueriesInParallel = cfg.FeatureToggles().IsEnabled("prometheusRunQueriesInParallel")
+	)
+
+	if hasPrometheusRunQueriesInParallel {
+		var (
+			m sync.Mutex
+		)
+
+		concurrentQueryCount, err := req.PluginContext.GrafanaConfig.ConcurrentQueryCount()
+		if err != nil {
+			logger := s.log.FromContext(ctx)
+			logger.Debug(fmt.Sprintf("Concurrent Query Count read/parse error: %v", err), "prometheusRunQueriesInParallel")
+			concurrentQueryCount = 10
+		}
 
-	for _, q := range req.Queries {
-		r := s.handleQuery(ctx, q, fromAlert, hasPromQLScopeFeatureFlag, hasPrometheusDataplaneFeatureFlag)
-		if r == nil {
-			continue
+		_ = concurrency.ForEachJob(ctx, len(req.Queries), concurrentQueryCount, func(ctx context.Context, idx int) error {
+			query := req.Queries[idx]
+			r := s.handleQuery(ctx, query, fromAlert, hasPromQLScopeFeatureFlag, hasPrometheusDataplaneFeatureFlag, hasPrometheusRunQueriesInParallel)
```

Comment:
```suggestion
			r := s.handleQuery(ctx, query, fromAlert, hasPromQLScopeFeatureFlag, hasPrometheusDataplaneFeatureFlag, true)
```

Reply:
The code reads weird especially for the false case when it passes the bool variable around. A hard coded true/false does the same job but more easy on the eyes.